File: javascript/lists/lookup.js
/**
 * List Lookup - X API v2
 * 
 * Endpoint: GET https://api.x.com/2/lists/:id
 * Docs: https://developer.x.com/en/docs/twitter-api/lists/list-lookup/api-reference/get-lists-id
 * 
 * Authentication: Bearer Token (App-only) or OAuth (User Context)
 * Required env vars: BEARER_TOKEN
 */

const needle = require('needle');

const token = process.env.BEARER_TOKEN;

// Replace with the list ID you want to look up
const listId = "84839422";
const endpointURL = `https://api.x.com/2/lists/${listId}`;

async function getRequest() {
    const params = {
        "list.fields": "created_at,follower_count,member_count,owner_id,description"
    };

    const res = await needle('get', endpointURL, params, {
        headers: {
            "User-Agent": "v2ListLookupJS",
            "authorization": `Bearer ${token}`
        }
    });

    if (res.body) {
        return res.body;
    } else {
        throw new Error('Unsuccessful request');
    }
}

(async () => {
    try {
        const response = await getRequest();
        console.dir(response, { depth: null });
    } catch (e) {
        console.log(e);
        process.exit(-1);
    }
    process.exit();
})();

File: python/bookmarks/lookup.py
"""
Bookmarks Lookup - X API v2
===========================
Endpoint: GET https://api.x.com/2/users/:id/bookmarks
Docs: https://developer.x.com/en/docs/twitter-api/tweets/bookmarks/api-reference/get-users-id-bookmarks

Authentication: OAuth 2.0 with PKCE (User Context)
Required env vars: BEARER_TOKEN (OAuth 2.0 user access token)
"""

import requests
import os
import json

# Note: This endpoint requires OAuth 2.0 User Context
# The bearer_token here should be the user's access token from OAuth 2.0 PKCE flow
bearer_token = os.environ.get("BEARER_TOKEN")


def create_url(user_id):
    return "https://api.x.com/2/users/{}/bookmarks".format(user_id)


def get_params():
    return {"tweet.fields": "created_at"}


def bearer_oauth(r):
    """
    Method required by bearer token authentication.
    """
    r.headers["Authorization"] = f"Bearer {bearer_token}"
    r.headers["User-Agent"] = "v2BookmarksLookupPython"
    return r


def connect_to_endpoint(url, params):
    response = requests.request("GET", url, auth=bearer_oauth, params=params)
    print(response.status_code)
    if response.status_code != 200:
        raise Exception(response.status_code, response.text)
    return response.json()


def main():
    # Replace with the authenticated user's ID
    user_id = "your-user-id"
    url = create_url(user_id)
    params = get_params()
    json_response = connect_to_endpoint(url, params)
    print(json.dumps(json_response, indent=4, sort_keys=True))


if __name__ == "__main__":
    main()
